.github/workflows/update-toc.yml

```diff
@@ -0,0 +1,35 @@
+name: Update TOC
+
+on:
+  push:
+    paths:
+      - README.md
+  pull_request:
+    paths:
+      - README.md
+
+jobs:
+  update-toc:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+      - name: Install markdown-toc
+        run: npm install -g markdown-toc
+      - name: Generate table of content
+        run: markdown-toc -i --maxdepth 3 README.md
+      - name: Create signed commit using GraphQL
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BRANCH: "${{ github.head_ref || github.ref_name }}"
+        run: |
+          echo "Creating commit on branch $BRANCH on repository $GITHUB_REPOSITORY"
+          gh api graphql \
```

Comment:
what if there are no changes to the table of contents, is there some automagic that avoids the commit in this case?

Reply:
Let me try that

Reply:
see https://github.com/zalando/logbook/actions/runs/15933771332/job/44948853745

Reply:
neat!